src/plugins/filters/filters.js

```diff
@@ -65,6 +59,17 @@ export class Filters extends BasePlugin {
     return PLUGIN_PRIORITY;
   }
 
+  /**
+   * Dependencies list.
+   *
+   * @type {Array}
+   */
+  dependecies = [
+    () => (this.hot.getPlugin('DropdownMenu') ? '' : 'DropdownMenu'),
+    () => (this.hot.getPlugin('HiddenRows') ? '' : 'HiddenRows'),
+    () => (hasCellType('checkbox') ? '' : 'CheckboxCellType'),
+  ];
+
```

Comment:
I'd move that complexity of the dependencies declaration to the abstraction (Base plugin). I think that right now, each function entry requires from plugin's developer to know how to program the module existence. Moreover, at first glance, it's not obvious why the function returns a dependency name or an empty string. 

As the plugin instances can be created multiple times (multiple Handsontable instances on a single page) I'd move the 
class property (`dependecies`) to static.

I'd suggest to simplify the deps declaration on the plugins such like:
```js
static get DEPS() {
  return [
    'plugin:DropdownMenu',
    'plugin:HiddenRows',
    'cell-type:checkbox',
  ];
}
```

Reply:
I like that ☝️

Reply:
The static field seems good. Although, `[module type]:[module name]` means we have to add additional logic to differentiate which module do we need (if `plugin` then do one thing, if `cell-type` do another thing) - and all things will be in imported/added to BasePlugin. Is it sth we want to see there?

Reply:
I need your review guys. I've prepared PoC within the https://github.com/handsontable/handsontable/pull/7430/commits/d9c05c45ba64a1e6e8c72fa475bbdf706da4c7c5. I've followed my suggestions. Additionally, the Base plugin aggregates the errors and throws them once when the plugins are initialized and right before the plugin enabling. 

For:
```js
registerPlugin(Filters);
registerPlugin(DropdownMenu);
registerPlugin(ContextMenu);
registerPlugin(CollapsibleColumns);
```
it looks like:
![Zrzut ekranu 2021-01-18 o 13 40 30](https://user-images.githubusercontent.com/571316/104917056-30e04200-5993-11eb-98b1-7de4fab46fbb.png)

Reply:
Looks good!

If BasePlugin is not the best place then we may move it to `core` and `updateSettings` after all plugins are registered and before we enable them? BasePlugin would not know about `hasEditor`, `hasCellType` helpers. Don't even have to know how many deps "types" are there as this is the `core` responsibility to verify modules and their deps.

On the other hand, a plugin is a valid container for other modules, implementing self-validation, as this is done now, is not wrong and offloads `core` from detailed plugin system implementation.

🤔

Reply:
I think that as long as it's for internal use we can restrict dependency up to the built-in modules, but in the future, it'd be worthy to have the possibility to add external dependencies.